h/api/search.py

```diff
@@ -31,7 +34,41 @@ def _match_clause_for_uri(uristr):
     }
 
 
-def build_query(request_params):
+def scan(es_client, query, fields):
```

Comment:
So

    search.scan(es_client=es_client, query=query, fields=[])

becomes

    request.es_client.scan(query=query, fields=[])

or in fact just

    es_client.scan(query=query, fields=[])

since `add_or_remove_nipsa()` would just be passed the `EsClient` instance not the whole request.